ssh/DOCS.md

```diff
@@ -1,55 +1,63 @@
+directory is mounted on the path `/config`.
 
-You can not run both variants at the same time. Enabling login via keys, will
-disable password login.
-
-1. Add a ssh key to  `authorized_keys` or set a `password` in the add-on configuration.
-2. Start the add-on.
-3. Connect to your device using your preferred SSH client and use `root` as
-   the username.
-
-After logging in, you will find yourself in this add-on’s container.
-The Home Assistant configuration directory is mounted on the path `/config`.
-
-Try it out using:
+This add-on comes bundled with [The Home Assistant CLI](https://www.home-assistant.io/hassio/commandline/). Try it out using:
 
 ```bash
 ha help
 ```
 
+### The In-Browser Terminal
+
+To enable the in-browser Terminal all you need to do is start the add-on, and click the Terminal menu item.
+
+### SSH Server Connection
+
+To connect using an SSH client, such as PuTTY, you need to supply additional configuration for this add-on. To enable SSH connectivity, you need to:
+
+- Provide authentication credentials - a password or SSH key(s)
+- Specify which TCP port to bind to, on the Home Assistant host
+
+You can then connect to the port specified, using the username `root`. Please note that enabling the SSH Server potentially makes your Home Assistant system less secure, as it might enable anyone on the internet to try to access your system. The security of your system also depends on your network set up, router settings, use of firewalls, etc. As a general recommendation, you should not activate this part of the add-on unless you understand the ramifications.
+
+If you enable connecting to the SSH Server using an SSH client, you are strongly recommended to use private/public keys to log in. As long as you keep the private part of your key safe, this makes your system much harder to break into. Using passwords is therefor generally considered a less secure mechanism. To generate private/public SSH keys, follow the [instructions for Windows][keygen-windows] and [these for other platforms][keygen].
+
+Enabling login via password, will disable key based login. You can not run both variants at the same time.
+
 ## Configuration
 
 Add-on configuration:
 
 ```yaml
 authorized_keys:
-  - ssh-rsa AKDJD3839...== my-key
+  - "ssh-rsa AKDJD3839...== my-key"
 password: ''
+server:
+  tcp_forwarding: false
 ```
 
 ### Option: `authorized_keys`
 
-Your **public keys** for the authorized key file. You can authorize multiple
-keys by adding multiple public keys to the list.
+Your **public keys** that you wish to accept for login. You can authorize multiple keys by adding multiple public keys to the list.
+
+If you get errors when adding your key, it is likely that the public key you're trying to add, contains characters that intervene with Yaml syntax. Try enclosing your key in double quotes to avoid this issue.
```

Comment:
```suggestion
Your **public keys** that you wish to accept for login. You can authorize multiple keys by adding multiple public keys to the list.

If you get errors when adding your key, it is likely that the public key you're trying to add, contains characters that intervene with YAML syntax. Try enclosing your key in double quotes to avoid this issue.
```

Reply:
I was initially hesitant about this, as a rule-of-thumb (in Swedish) is that you only uppercase it if you would pronounce the acronym letter by letter, like Y A M L.

I read up on this a bit and it seems for English, it's all about what the reader is familiar with. There are two other uses of the uppercase spelling (`YAML`) in this project's documentation, so it makes absolutely sense to stick to that.